roles/openshift_health_checker/openshift_checks/disk_availability.py

```diff
@@ -81,6 +100,33 @@ def run(self, tmp, task_vars):
             config_bytes = max(config.get(name, 0) for name in group_names) * 10**9
             recommended_bytes = config_bytes or recommended_bytes
 
+            # if an "upgrade" context is set, lower the minimum disk requirement
+            # as this signifies an in-place upgrade - the node might have the
+            # required total disk space, but some of that space may already be
+            # in use by the existing OpenShift deployment.
+            context = get_var(task_vars, "r_openshift_health_checker_playbook_context", default="")
+            if context == "upgrade":
+                min_upgrade_rec = self.recommended_disk_upgrade_bytes.get(path, {})
+                if min_upgrade_rec:
+                    min_upgrade_rec = max(min_upgrade_rec.get(name, 0) for name in group_names)
+                else:
+                    # if not able to resolve a recommended value, default to non-upgrade recommendation
+                    min_upgrade_rec = max(recommendation.get(name, 0) for name in group_names)
+
+                if free_bytes < min_upgrade_rec:
```

Comment:
I feel like this duplicates logic unnecessarily. Couldn't we simple use the upgrade values when it's an upgrade and then use the same code paths to compare free / required?